### PR TITLE
[Crane] Remove bad SideEffect pattern

### PR DESCRIPTION
**Pull request by manuelvicnt**

### CODE REVIEW

File: Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt [hunk 1]
```diff
@@ -102,20 +108,46 @@ class DetailsActivity : ComponentActivity() {
...
+        }
     }
 
-    SideEffect {
-        if (cityDetailsResult is Result.Error) {
-            onErrorLoading()
+    Crossfade(targetState = uiState) { currentUiState ->
+        when {
+            currentUiState.exploreModel != null -> {
+                DetailsContent(currentUiState.exploreModel, modifier)
```
Comment (chrisbanes) [inline Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt:138]:
```suggestion
                DetailsContent(currentUiState.exploreModel, Modifier.fillMaxSize())
```

File: Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt [hunk 2]
```diff
@@ -102,20 +108,46 @@ class DetailsActivity : ComponentActivity() {
...
+            DetailsScreenUiState(cityDetailsResult.data)
+        } else {
+            DetailsScreenUiState(throwError = true)
+        }
     }
 
-    SideEffect {
-        if (cityDetailsResult is Result.Error) {
-            onErrorLoading()
+    Crossfade(targetState = uiState) { currentUiState ->
```
Comment (chrisbanes) [inline Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt:135]: pass `modifier` to the Crossfade

File: Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt [hunk 3]
```diff
@@ -102,20 +108,46 @@ class DetailsActivity : ComponentActivity() {
...
-    val cityDetailsResult = remember(viewModel) { viewModel.cityDetails }
-    if (cityDetailsResult is Result.Success<ExploreModel>) {
-        DetailsContent(cityDetailsResult.data, modifier)
+    val uiState by produceState(
+        key1 = viewModel,
+        initialValue = DetailsScreenUiState(isLoading = true)
+    ) {
+        val cityDetailsResult = viewModel.cityDetails
+        value = if (cityDetailsResult is Result.Success<ExploreModel>) {
+            DetailsScreenUiState(cityDetailsResult.data)
```
Comment (chrisbanes) [inline Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt:129]: Should the mapping to `DetailsScreenUiState` be done in the VM?

**Review by chrisbanes — APPROVED** [review]

File: Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt (hunk 3 above)
Comment (manuelvicnt) [reply Crane/app/src/main/java/androidx/compose/samples/crane/details/DetailsActivity.kt:131]: Added comment explaining why. Thanks!

**Pull request merged**